### PR TITLE
Add `quickwit split mark-for-deletion` command

### PR DESCRIPTION
**Pull request by guilload**
### Description
Add `quickwit split mark` command

### How was this PR tested?
Add unit test

### CODE REVIEW

File: quickwit-cli/src/split.rs [hunk 1]
```diff
@@ -89,6 +89,18 @@ pub fn build_split_command<'a>() -> Command<'a> {
                         .required(false),
                 ])
             )
+        .subcommand(
+            Command::new("delete")
+                .about("Removes one or multiple splits from an index.")
+                .args(&[
+                    arg!(--config <CONFIG> "Quickwit config file").env("QW_CONFIG"),
+                    arg!(--index <INDEX> "ID of the target index"),
+                    arg!(--splits <SPLITS> "Comma-separated list of split IDs.")
```
Comment (guilload) [inline quickwit-cli/src/split.rs:98]: Really, this should be `<SPLIT IDs>` but clap is not happy with whitespace here.

File: quickwit-cli/src/split.rs [hunk 2]
```diff
@@ -291,6 +336,25 @@ async fn list_split_cli(args: ListSplitArgs) -> anyhow::Result<()> {
...
+        .resolve(&quickwit_config.metastore_uri())
+        .await?;
+    let split_ids: Vec<&str> = args
+        .split_ids
+        .iter()
+        .map(|split_id| split_id.as_ref())
+        .collect();
+    metastore
+        .mark_splits_for_deletion(&args.index_id, &split_ids)
+        .await?;
```
Comment (guilload) [inline quickwit-cli/src/split.rs:354]: Note that we don't run GC here, so "delete" here is misleading. Some options:
- rename the option `--mark-as-deleted`
- run the GC
- run the GC and add a `--no-gc` option
- your awesome idea here :)

Let me know what you think and I'll make the appropriate changes.

**Comment by guilload** [comment]
PTAL

File: quickwit-cli/src/split.rs (hunk 2 above)
Comment (guilload) [reply quickwit-cli/src/split.rs:354]: I'm going to rename the command `mark-for-deletion` and provide a `mark` alias. The gc will delete the splits later. One can always run the gc manually with `quickwit index gc`  and a short grace period if necessary.

**Review by fmassot — APPROVED** [review]

**Pull request merged**